pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,129 @@
+	// be empty or match the first entry in the list.
+	// Default is 10.0.0.0/16 for all platforms other than libvirt.
+	// For libvirt, the default is 192.168.126.0/24.
+	//
+	// +optional
+	MachineNetwork []MachineNetworkEntry `json:"machineNetwork,omitempty"`
+
+	// ClusterNetwork is the list of IP address pools for pods.
+	// Default is 10.128.0.0/14 and a host prefix of /23.
+	//
+	// +optional
+	ClusterNetwork []ClusterNetworkEntry `json:"clusterNetwork,omitempty"`
+
+	// ServiceNetwork is the list of IP address pools for services.
+	// Default is 172.30.0.0/16.
+	// NOTE: currently only one entry is supported.
+	//
+	// +kubebuilder:validation:MaxItems=1
+	// +optional
+	ServiceNetwork []string `json:"serviceNetwork,omitempty"`
+}
+
+// MachineNetworkEntry is a single IP address block for node IP blocks.
+type MachineNetworkEntry struct {
+	// CIDR is the IP block address pool for machines within the cluster.
+	CIDR string `json:"cidr"`
+}
+
+// ClusterNetworkEntry is a single IP address block for pod IP blocks. IP blocks
+// are allocated with size 2^HostSubnetLength.
+type ClusterNetworkEntry struct {
+	// CIDR is the IP block address pool.
+	CIDR string `json:"cidr"`
+
+	// HostPrefix is the prefix size to allocate to each node from the CIDR.
+	// For example, 24 would allocate 2^8=256 adresses to each node. If this
+	// field is not used by the plugin, it can be left unset.
+	// +optional
+	HostPrefix int32 `json:"hostPrefix,omitempty"`
+}
+
+type ClusterProgressInfo struct {
+	// progress info
+	ProgressInfo string `json:"progressInfo,omitempty"`
+	// Time at which the cluster install progress was last updated.
+	LastProgressUpdateTime *metav1.Time `json:"lastProgressUpdateTime,omitempty"`
+}
+
+type HostNetwork struct {
+	// cidr
+	Cidr string `json:"cidr,omitempty"`
+	// host ids
+	HostIds []string `json:"hostIDs"`
+}
+
+// ClusterStatus defines the observed state of Cluster
+type ClusterStatus struct {
+	State                        string              `json:"state,omitempty"`
+	StateInfo                    string              `json:"stateInfo,omitempty"`
+	HostNetworks                 []HostNetwork       `json:"hostNetworks,omitempty"`
+	InstallationStartTime        string              `json:"installationStartTime,omitempty"`
+	InstallationCompletionTime   *metav1.Time        `json:"installationCompletionTime,omitempty"`
+	Hosts                        int                 `json:"hosts,omitempty"`
+	Progress                     ClusterProgressInfo `json:"progress,omitempty"`
+	ValidationsInfo              string              `json:"validationsInfo,omitempty"`
+	ConnectivityMajorityGroups   string              `json:"connectivityMajorityGroups,omitempty"`
+	LastUpdateTime               *metav1.Time        `json:"lastUpdateTime,omitempty"`
+	ControllerLogsCollectionTime *metav1.Time        `json:"controllerLogsCollectionTime,omitempty"`
+	// Display api errors
+	Error string `json:"error,omitempty"`
```

Comment:
Between State, StateInfo, Progress, ValidationsInfo, and Error, I am not entirely sure how to represent status cleanly. Would it be possible to get a good example of a serialized resource from the REST API that shows how all of these are used? Would help beat this down into something smaller, I feel like there's too much going on here for a caller to check if things are ok or not. 

ValidationsInfo seems like it would belong on Agents not the cluster?

Reply:
State is just the state in the state machine.
StateInfo is just a string with some extra information regarding that state that could be helpful to the user.
Progress is used only during the actual installation part to report on the installation progress.
ValidationsInfo is used only during the discovery phase (before the actual installation part).
Error is something that we added for the k8s API to display API errors.

Reply:
We could combine the first two into a condition, Type = AgentInstallState, Reason = your set of supported states, Message = StateInfo.

Progress also feels like a condition.

ValidationsInfo feels like it belongs on Agents possibly? Or if cluster wide then another condition.

I'm unsure where or how Error fits in, or why it wouldn't fall into one of the previous.

Reply:
What are the criteria for condition vs. property in status?  For example, it looks like in BMH I can find the provisioning state in the status.  (I have no problem moving things, just trying to learn)
As for the validations, these are validations at the cluster level (e.g., if required cluster parameters are defined and valid, has enough masters, etc.).  We have other validations on agents.

Reply:
400-500 errors from the assisted service backend.

Condition: Type: AssistedServiceError, Reason: Error, Message: 504 error received

Reply:
We hope to eliminate this condition in the future.